backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/exceptions/DocumentProcessingException.java

```diff
@@ -10,57 +10,54 @@
  * it under the terms of the GNU General Public License as
  * published by the Free Software Foundation, either version 2 of the
  * License, or (at your option) any later version.
- * 
+ *
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
  * GNU General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU General Public
  * License along with this program. If not, see
  * <http://www.gnu.org/licenses/gpl-2.0.html>.
  * #L%
  */
 
-import org.adempiere.exceptions.AdempiereException;
-
 import de.metas.document.engine.IDocument;
 import de.metas.document.engine.IDocumentBL;
 import de.metas.i18n.ITranslatableString;
 import de.metas.i18n.TranslatableStringBuilder;
 import de.metas.i18n.TranslatableStrings;
 import de.metas.util.Check;
 import de.metas.util.Services;
+import org.adempiere.exceptions.AdempiereException;
+
+import javax.annotation.Nullable;
 
 /**
- * Exception thrown when document processing failed.
- * 
- * @author tsa
+ * Miscellaneous document processing failure
  *
+ * @author tsa
  */
 @SuppressWarnings("serial")
 public class DocumentProcessingException extends AdempiereException
 {
-	public DocumentProcessingException(final IDocument document, final String docAction, final Throwable cause)
-	{
-		super(buildMsg((String)null, document, docAction, cause), cause);
-		putParametetersFrom(cause);
-	}
-
 	public DocumentProcessingException(final IDocument document, final String docAction)
 	{
-		super(buildMsg((String)null, document, docAction, (Throwable)null));
+		super(buildMsg(null, document, docAction));
 	}
 
 	public DocumentProcessingException(final String message, final Object documentObj, final String docAction)
 	{
-		super(buildMsg(message, documentObj, docAction, (Throwable)null));
+		super(buildMsg(message, documentObj, docAction));
 	}
 
-	private static final ITranslatableString buildMsg(final String message, final Object documentObj, final String docAction, final Throwable cause)
+	private static ITranslatableString buildMsg(
+			@Nullable final String message,
+			@Nullable final Object documentObj,
+			@Nullable final String docAction)
 	{
 		final TranslatableStringBuilder msg = TranslatableStrings.builder();
-		if (Check.isEmpty(message, true))
+		if (message == null || Check.isBlank(message))
```

Comment:
yeah... you will see it in several places... it's because of IntelliJ null analysis which keeps complaining about a possible null problem.
not sure who is better...